src/index.js

```diff
@@ -51,20 +94,32 @@ async function purgeOuter(host, path, log, exact) {
     if (!res.ok) {
       throw new Error(msg);
     }
+    results.push({ status: 'ok', url });
   } catch (e) {
     log.error('Unable to purge outer CDN', e);
     return { status: 'error', url };
   }
   if (!exact) {
-    if (path.endsWith('.html')) {
-      // if .html extension, also purge URL without it
-      await purgeOuter(host, path.substring(0, path.lastIndexOf('.')), log, true);
-    } else if (!path.split('/').pop().includes('.')) {
-      // if no extension, also purge URL with .html extension
-      await purgeOuter(host, `${path}.html`, log, true);
+    const file = path.split('/').pop();
+    if (!file) {
+      // directory, also purge index(.html)
+      results.push(await purgeOuter(host, `${path}index`, log, true));
+      results.push(await purgeOuter(host, `${path}index.html`, log, true));
+    } else {
+      if (file.match(/index\.?/)) {
```

Comment:
i'd prefer `file == 'index' || file == 'index.html'`

Reply:
What about `index.plain.html` that is sometimes embedded?